### PR TITLE
Spaces: Use Spaces API to create a Space in Scenarios

### PR DESCRIPTION
**Pull request by zspencer**
This swaps our scenarios to append test data to the local database, rather than re-use the `system-test` space over and over and over again.

It runs safely on CI, but if you run the test suite twice you start to see errors.

I believe this is because we're not applying a the `testId` to the Invitation Email, so when an invitation goes out it goes out to the same email address each time; which can result in race-conditions when following links.

It may be worth tilting at that windmill a bit more before merging; but I'm not sure.

Thoughts?

### CODE REVIEW

File: features/support/env.js [hunk 1]
```diff
@@ -14,6 +15,10 @@ class CustomWorld {
   constructor() {
     this.driver = driver;
   }
+
+  testId() {
```
Comment (zspencer) [inline features/support/env.js:19]: There may already be something like this in cucumber, but I didn't take the time to find out :X

File: app/models/space/serializer.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+# Serializes {Space}s for programmatic consumption
+class Space::Serializer < ApplicationSerializer
+  # @return [Space]
+  alias space resource
+
+  def to_json(*_args)
```
Comment (anaulin) [inline app/models/space/serializer.rb:8]: Feels like it is time to start introducing https://github.com/jsonapi-serializer/jsonapi-serializer (or similar), making this all a little better. Maybe in a follow-up PR.

File: features/lib/Api.js [hunk 3]
```diff
@@ -1,38 +1,57 @@
-const axios = require('axios');
+const axios = require("axios");
+const { camelCase } = require('change-case');
+const applyCaseMiddleware = require("axios-case-converter").default;
```
Comment (anaulin) [inline features/lib/Api.js:3]: Maybe run an autoformatter to get all these quotes consistent.

**Review by anaulin — APPROVED** [review]
I'm fine with getting this merged in even if right now there are issues running the suite twice. Hopefully we will address that sooner rather than later.

File: features/lib/Api.js (hunk 3 above)
Comment (zspencer) [reply features/lib/Api.js:3]: Done!

File: app/models/space/serializer.rb (hunk 2 above)
Comment (zspencer) [reply app/models/space/serializer.rb]: Concur.

**Pull request merged**